go/cmd/server/logging/logging.go

```diff
@@ -0,0 +1,306 @@
+
+	fields := make([]arrow.Field, 0)
+	columns := make([]array.Interface, 0)
+	for _, entityName := range fcoSchema.Entities {
+		protoArr := entityNameToEntityValues[entityName]
+		if len(protoArr) == 0 {
+			break
+		}
+		valArrowArray, err := gotypes.ProtoValuesToArrowArray(protoArr, arrowMemory, len(columnNameToProtoValueArray))
+		if err != nil {
+			return nil, err
+		}
+		arrowType, err := gotypes.ValueTypeEnumToArrowType(fcoSchema.EntityTypes[entityName])
+		if err != nil {
+			return nil, err
+		}
+		fields = append(fields, arrow.Field{
+			Name: entityName,
+			Type: arrowType,
+		})
+		columns = append(columns, valArrowArray)
+	}
+
+	for _, featureName := range fcoSchema.Features {
+
+		protoArr := columnNameToProtoValueArray[featureName]
+		if len(protoArr) == 0 {
+			break
+		}
+		arrowArray, err := gotypes.ProtoValuesToArrowArray(protoArr, arrowMemory, len(columnNameToProtoValueArray))
+		if err != nil {
+			return nil, err
+		}
+
+		arrowType, err := gotypes.ValueTypeEnumToArrowType(fcoSchema.FeaturesTypes[featureName])
+
+		if err != nil {
+			return nil, err
+		}
+		fields = append(fields, arrow.Field{
+			Name: featureName,
+			Type: arrowType,
+		})
+		columns = append(columns, arrowArray)
+	}
+	schema := arrow.NewSchema(
+		fields,
+		nil,
+	)
+
+	result := array.Record(array.NewRecord(schema, columns, int64(len(memoryBuffer.logs))))
+
+	tbl := array.NewTableFromRecords(schema, []array.Record{result})
+	return array.Table(tbl), nil
+}
+
+type Schema struct {
+	Entities      []string
+	Features      []string
+	EntityTypes   map[string]types.ValueType_Enum
+	FeaturesTypes map[string]types.ValueType_Enum
+}
+
+func GetSchemaFromFeatureService(featureService *model.FeatureService, entities []*model.Entity, featureViews []*model.FeatureView, onDemandFeatureViews []*model.OnDemandFeatureView) (*Schema, error) {
+	fvs := make(map[string]*model.FeatureView)
+	odFvs := make(map[string]*model.OnDemandFeatureView)
+
+	entityNames := make([]string, 0)
+	for _, entity := range entities {
+		entityNames = append(entityNames, entity.JoinKey)
+	}
+
+	for _, featureView := range featureViews {
+		fvs[featureView.Base.Name] = featureView
+	}
+
+	for _, onDemandFeatureView := range onDemandFeatureViews {
+		odFvs[onDemandFeatureView.Base.Name] = onDemandFeatureView
+	}
+
+	entityJoinKeyToType := make(map[string]types.ValueType_Enum)
+	for _, entity := range entities {
+		entityJoinKeyToType[entity.JoinKey] = entity.ValueType
+	}
+
+	allFeatureTypes := make(map[string]types.ValueType_Enum)
+	features := make([]string, 0)
+	for _, featureProjection := range featureService.Projections {
+		// Create copies of FeatureView that may contains the same *FeatureView but
+		// each differentiated by a *FeatureViewProjection
+		featureViewName := featureProjection.Name
+		if fv, ok := fvs[featureViewName]; ok {
+			for _, f := range fv.Base.Features {
+				// add feature to map
+				features = append(features, f.Name)
+				allFeatureTypes[f.Name] = f.Dtype
```

Comment:
feature name is not unique within feature service. it's unique only within its feature view. So would be good idea to use full feature name here `{view.name}__{feature.name}`

Reply:
Done, but I also had to specify fullfeaturenames = true when querying since other wise the returned feature names don't actually have all of the information I need.